src/pyfaf/ureport.py

```diff
@@ -531,17 +533,81 @@ def save_attachment(db, attachment):
         log.warning("Unknown attachment type")
 
 
+def valid_known_type(known_type):
+    """
+    Check if all "known" values from configuration file are correct
+    allowed_known_type is list with allowed values
+    """
+    allowed_known_type = ['EQUAL_UREPORT_EXISTS', 'BUG_OS_MINOR_VERSION',
+                          'BUG_OS_MAJOR_VERSION']
+
+    for ktype in known_type:
+        if ktype not in allowed_known_type and ktype.strip() != "":
+            log.error("Known type '{0}' is not supported by FAF Server"
+                      .format(ktype))
+            return False
+
+    return True
+
+
 def is_known(ureport, db, return_report=False, opsysrelease_id=None):
     ureport = ureport2(ureport)
 
     problemplugin = problemtypes[ureport["problem"]["type"]]
     report_hash = problemplugin.hash_ureport(ureport["problem"])
 
-    report = get_report_by_hash(db, report_hash)
+    known_type = []
+
+    # Split allowed types from config
+    if 'ureport.known' in config and config['ureport.known'].strip() != "":
+        known_type = config['ureport.known'].strip().split(" ")
+
+    if len(known_type) > 0 and not valid_known_type(known_type):
+        return None
+
+    report_os = {'name':None,
+                 'version':None,
+                 'architecture':None}
+
+    if 'EQUAL_UREPORT_EXISTS' in known_type:
+        report_os = ureport["os"]
+
+    reports = db.session.query(Report).all()
+
+
+    report = find_report(db, report_hash, os_name=report_os['name'], os_version=report_os['version'], os_arch=report_os['architecture'])
 
     if report is None:
         return None
-    if get_reportbz(db, report.id, opsysrelease_id).first() is not None:
+
+    if 'EQUAL_UREPORT_EXISTS' in known_type and report is not None:
```

Comment:
report is not None is redundant in all of these, it's already tested on line 580.